identity-hub-did/src/test/java/org/eclipse/dataspaceconnector/identityhub/did/IdentityHubCredentialsVerifierTest.java

```diff
@@ -0,0 +1,175 @@
+                .service(List.of(new Service("IdentityHub", "IdentityHub", hubBaseUrl))).build();
+        var credential = generateVerifiableCredential();
+        var jws = buildSignedJwt(credential, issuer);
+        when(identityHubClient.getVerifiableCredentials(hubBaseUrl))
+                .thenReturn(StatusResult.success(List.of(jws)));
+        when(signatureVerifier.isSignedByIssuer(jws)).thenReturn(false);
+
+        // Act
+        var credentials = credentialsVerifier.getVerifiedCredentials(didDocument);
+
+        // Assert
+        assertThat(credentials.succeeded());
+        assertThat(credentials.getContent().size()).isEqualTo(0);
+    }
+
+    @Test
+    public void getVerifiedClaims_hubUrlNotResolved() {
+        // Arrange
+        var didDocument = DidDocument.Builder.newInstance().build();
+
+        // Act
+        var credentials = credentialsVerifier.getVerifiedCredentials(didDocument);
+
+        // Assert
+        assertThat(credentials.failed());
+    }
+
+    @Test
+    public void getVerifiedClaims_idHubCallFails() {
+
+        // Arrange
+        var didDocument = DidDocument.Builder.newInstance().service(List.of(new Service("IdentityHub", "IdentityHub", hubBaseUrl))).build();
+        when(identityHubClient.getVerifiableCredentials(hubBaseUrl)).thenReturn(StatusResult.failure(ResponseStatus.FATAL_ERROR));
+
+        // Act
+        var credentials = credentialsVerifier.getVerifiedCredentials(didDocument);
+
+        // Assert
+        assertThat(credentials.failed());
+    }
+
+    @Test
+    public void getVerifiedClaims_verifiableCredentialsWithWrongFormat() {
+
+        // Arrange
+        var didDocument = DidDocument.Builder.newInstance()
+                .service(List.of(new Service("IdentityHub", "IdentityHub", hubBaseUrl))).build();
+        var jws = new SignedJWT(new JWSHeader.Builder(JWSAlgorithm.ES256).build(), new JWTClaimsSet.Builder().build());
+        when(identityHubClient.getVerifiableCredentials(hubBaseUrl)).thenReturn(StatusResult.success(List.of(jws)));
+        when(signatureVerifier.isSignedByIssuer(jws)).thenReturn(true);
+
+        // Act
+        var credentials = credentialsVerifier.getVerifiedCredentials(didDocument);
+
+        // Assert
+        assertThat(credentials.succeeded());
+        assertThat(credentials.getContent().isEmpty());
+    }
+
+    @Test
+    public void getVerifiedClaims_verifiableCredentialsWithMissingId() {
+
+        // Arrange
+        var didDocument = DidDocument.Builder.newInstance()
+                .service(List.of(new Service("IdentityHub", "IdentityHub", hubBaseUrl))).build();
+        var jwsHeader = new JWSHeader.Builder(JWSAlgorithm.ES256).build();
+        var jwtClaims = new JWTClaimsSet.Builder()
+                .claim("vc", Map.of(FAKER.lorem().word(), FAKER.lorem().word()))
+                .issuer(FAKER.lorem().word())
+                .audience(VC_AUDIENCE)
+                .subject("verifiable-credential")
```

Comment:
From the examples in https://www.w3.org/TR/vc-data-model/#example-usage-of-the-id-property it seems the subject should be the DID of the subject (i.e. same value as `credentialSubject.id`)